apps/src/dance/ai/DanceAiModal.tsx

```diff
@@ -758,58 +640,16 @@ const DanceAiModal: React.FunctionComponent<DanceAiModalProps> = ({
         {(mode === Mode.GENERATING ||
           mode === Mode.GENERATED ||
           mode === Mode.RESULTS) && (
-          <div
-            id="preview-area"
-            className={classNames(moduleStyles.previewArea, previewAreaClass)}
-          >
-            <div id="flip-card" className={moduleStyles.flipCard}>
-              <div
-                id="flip-card-inner"
-                className={classNames(
-                  moduleStyles.flipCardInner,
-                  mode === Mode.RESULTS &&
-                    currentToggle === Toggle.CODE &&
-                    moduleStyles.flipCardInnerFlipped
-                )}
-              >
-                <div
-                  id="flip-card-front"
-                  className={moduleStyles.flipCardFront}
-                >
-                  {indexesToPreview.map(index => {
-                    return (
-                      <div
-                        id={'preview-container-' + index}
-                        key={'preview-container-' + index}
-                        className={moduleStyles.previewContainer}
-                        style={{
-                          animationDuration: previewAppearDuration + 'ms',
-                        }}
-                      >
-                        <AiVisualizationPreview
-                          id={'ai-preview-' + index}
-                          code={getPreviewCode(getGeneratedEffect(index))}
-                          size={previewSize}
-                        />
-                      </div>
-                    );
-                  })}
-                </div>
-                <div id="flip-card-back" className={moduleStyles.flipCardBack}>
-                  {mode === Mode.RESULTS && (
-                    <div
-                      id="block-preview"
-                      className={moduleStyles.blockPreview}
-                    >
-                      {currentGeneratedEffect && (
-                        <AiBlockPreview results={currentGeneratedEffect} />
-                      )}
-                    </div>
-                  )}
-                </div>
-              </div>
-            </div>
-          </div>
+          <DanceAiModalFlipCard
+            mode={mode}
+            generatingProgressStep={generatingProgress.step}
+            badGeneratedResultsCount={BAD_GENERATED_RESULTS_COUNT}
+            currentToggle={currentToggle}
+            previewAppearDuration={previewAppearDuration}
+            currentGeneratedEffect={currentGeneratedEffect}
+            getPreviewCode={getPreviewCode}
+            getGeneratedEffect={getGeneratedEffect}
+          />
```

Comment:
This component is a candidate for further refactoring, since it's pulling in a lot of state from the Modal and I think several of these things can be crushed together into better structures. But that's a future item.